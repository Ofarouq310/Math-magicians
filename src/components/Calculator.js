import React from 'react';
import calculate from '../logic/calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: 0,
      next: null,
      operation: null,
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(e) {
    // eslint-disable-next-line react/no-access-state-in-setstate
    this.setState(calculate(this.state, e.target.innerHTML));
  }

  render() {
    const { total, next, operation } = this.state;
    return (
      <div className="calculator">
        <div id="calculator-display">
          {total}
          {operation}
          {next}
        </div>
        <button type="button" onClick={this.handleClick} className="calculator-key" id="key-clear">AC</button>
        <button type="button" onClick={this.handleClick} className="calculator-key" id="key-sign">+/-</button>
        <button type="button" onClick={this.handleClick} className="calculator-key" id="key-percent">%</button>
        <button type="button" onClick={this.handleClick} className="calculator-key orange-key" id="key-division">รท</button>
        <button type="button" onClick={this.handleClick} className="calculator-key" id="key-7">7</button>
        <button type="button" onClick={this.handleClick} className="calculator-key" id="key-8">8</button>
        <button type="button" onClick={this.handleClick} className="calculator-key" id="key-9">9</button>
        <button type="button" onClick={this.handleClick} className="calculator-key orange-key" id="key-multiply">x</button>
        <button type="button" onClick={this.handleClick} className="calculator-key" id="key-4">4</button>
        <button type="button" onClick={this.handleClick} className="calculator-key" id="key-5">5</button>
        <button type="button" onClick={this.handleClick} className="calculator-key" id="key-6">6</button>
        <button type="button" onClick={this.handleClick} className="calculator-key orange-key" id="key-minus">-</button>
        <button type="button" onClick={this.handleClick} className="calculator-key" id="key-1">1</button>
        <button type="button" onClick={this.handleClick} className="calculator-key" id="key-2">2</button>
        <button type="button" onClick={this.handleClick} className="calculator-key" id="key-3">3</button>
        <button type="button" onClick={this.handleClick} className="calculator-key orange-key" id="key-plus">+</button>
        <button type="button" onClick={this.handleClick} className="calculator-key" id="key-0">0</button>
        <button type="button" onClick={this.handleClick} className="calculator-key" id="key-dot">.</button>
        <button type="button" onClick={this.handleClick} className="calculator-key orange-key" id="key-equals">=</button>
      </div>
    );
  }
}
export default Calculator;
