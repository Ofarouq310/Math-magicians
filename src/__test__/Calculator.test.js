import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Calculator from '../components/Calculator';

describe('check Calculator', () => {
  const tree = renderer.create(<Router><Calculator /></Router>).toJSON();
  test('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});
