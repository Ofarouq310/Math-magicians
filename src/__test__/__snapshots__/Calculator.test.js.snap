// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check Calculator renders correctly 1`] = `
<div
  className="main-container"
>
  <div
    id="header"
  >
    <h1>
      Let's do some math!
    </h1>
  </div>
  <div
    className="calculator"
  >
    <div
      id="calculator-display"
    >
      0
    </div>
    <button
      className="calculator-key"
      id="key-clear"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="calculator-key"
      id="key-sign"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="calculator-key"
      id="key-percent"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="calculator-key orange-key"
      id="key-division"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="calculator-key"
      id="key-7"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="calculator-key"
      id="key-8"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="calculator-key"
      id="key-9"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="calculator-key orange-key"
      id="key-multiply"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="calculator-key"
      id="key-4"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="calculator-key"
      id="key-5"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="calculator-key"
      id="key-6"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="calculator-key orange-key"
      id="key-minus"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="calculator-key"
      id="key-1"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="calculator-key"
      id="key-2"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="calculator-key"
      id="key-3"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="calculator-key orange-key"
      id="key-plus"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="calculator-key"
      id="key-0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="calculator-key"
      id="key-dot"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="calculator-key orange-key"
      id="key-equals"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`check Navbar renders correctly 1`] = `
<nav>
  <h1>
    Math Magicians
    <span>
       
    </span>
  </h1>
  <ul
    className="navLink"
  >
    <li
      className="navItem"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <span>
          Home
        </span>
      </a>
    </li>
    <span>
      |
    </span>
    <li
      className="navItem"
    >
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <span>
      |
    </span>
    <li
      className="navItem"
    >
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
